internal/command/junit/junit.go

```diff
@@ -206,31 +218,41 @@ func junitXMLTestReport(suite *moduletest.Suite, sources map[string][]byte) ([]b
 					// why the test was skipped?
 				}
 			case moduletest.Fail:
+				var failedAssertion tfdiags.Diagnostic
+				for key, d := range diagsMap {
+					// Find the diag resulting from a failed assertion
+					if d.Description().Summary == failedTestSummary {
```

Comment:
Nitpick but could we not filter the diagnostics based on `ExtraInfo` instead here? We already use that concept in other places, see for example: https://github.com/hashicorp/terraform/blob/a8b126aeffdd1876da51e59a7cc685ad030e86d3/internal/tfdiags/diagnostic_extra.go#L142-L148

Reply:
I didn't know that was an option, thanks! I'll experiment and push an update